### ISSUE TITLE
[Issue] Changing "nestedDepth" creates massive log spam and game instability 

### ISSUE THREAD

**Issue by Peca21** [nanite/Compressium#15]
### Minecraft Version

1.19.2

### Modloader

43.2.3

### Mod version

Compressium-2.1.1-build.8+mc1.19.jar

### Repeatable?

- [X] Yes

### Link to log files

[2023-01-14-10.log.gz](https://github.com/nanite/Compressium/files/10420298/2023-01-14-10.log.gz)
Log is 30MB so here is a compressed file

### Steps to reproduce the issue

1. Change "nestedDepth" to something lower than 9 (I used 3)
2. Start Minecraft

### Description of the issue you are having

Massive log spam after changing "nestedDepth" to 3 and (but not sure about this one) missing tags like #minecraft:mineable/pickaxe and #minecraft:mineable/shovel making shovels and pickaxes useless.

About that second problem:
I created a modpack and was testing ore generation and realized that I couldn't mine ores with any pickaxe. The one probe reported them all as "No tool" needed. Same thing with dirt, sand and gravel, no tool needed. I wanted to look at the logs to see what was the problem, but the log had like 100k lines in it and a lot of them were mentioning Compressum block recipe loading (Visible in the log I provided above). So to more easily troubleshoot I decided to set "nestedDepth"to 9 again to hopefully disable the log spam. And it worked, no more log spam after setting it ot 9. 

BUT my pickaxe and shovel started working again after changing "nestedDepth" to 9. Not sure how or why is the log spam breaking minecraft block tags, but it looks like it is. This also may be a forge problem, not the problem in your mod but I am not sure.

So basically setting "nestedDepth" to 3 (or anything lower than 9):
1. Creates Massive log spam
2. Breaks some vanilla minecraft tags

### Information

- [X] I have provided as much information as possible
